PictureViewMaster/PictureMasterView/PictureMasterViewController.swift

```diff
@@ -246,11 +261,35 @@ public class PictureMasterViewController: UIViewController , UIGestureRecognizer
     private func resetViewFrame(view: UIView, animated: Bool, completion: ((Bool) -> Void)?) {
         let duration = animated ? 0.2 : 0.0
         UIView.animateWithDuration(duration, delay: 0.0, options: .CurveEaseOut, animations: {
-            let imageView = view as! UIImageView
-            view.frame = self.originalImageViewFitFrameForImage(imageView.image!)
+            let imageView = view
+            view.frame = self.minimumSizeForView(imageView)
+//              view.frame = self.originalImageViewFitFrameForImage(imageView.image!)
             }, completion:completion)
     }
     
+    private func minimumSizeForView(view: UIView) -> CGRect {
+        let screenSize = UIScreen.mainScreen().bounds
+        var frame = CGRect()
+        
+        let heightRatio = view.frame.size.height / view.frame.size.width
+        let widthRatio = view.frame.size.width / view.frame.size.height
+        let screenRatio = screenSize.width / screenSize.height
+        
```

Comment:
`let size: CGSize`

Inside the `if` you define
`size = this`
else
`size = that`